components/views/login/index.tsx

```diff
@@ -0,0 +1,70 @@
+import { useState, FormEvent } from 'react'
+import Link from 'next/link'
+
+import { FormProps } from '../../../utilities/types/formTypes'
+import styles from './Login.module.scss'
+import useForm from '../../../utilities/hooks/useForm'
+
+const INITIAL_STATE = {
+    email: '',
+    password: ''
+} as FormProps
+
+const INITIAL_ERROR = {
+    message: 'Enter message',
+    hasError: false
+} as { message: string; hasError: boolean }
+
+const Login = () => {
+    const { input, handleChange, handleReset } = useForm(INITIAL_STATE)
+    const [error, setError] = useState(INITIAL_ERROR)
+
+    const { email, password } = input
```

Comment:
You can destructure these 2 where `input (which shortly will be form)` is called on line 19.

```
const {form: { email, password }, handleChange, handleReset } = useForm(INITIAL_STATE)
```